nemo/collections/tts/models/speechllm/megatron_t5_speechllm_model.py

```diff
@@ -203,7 +203,14 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
 
         self.frozen_model.enc_dec_model.attn_prior_scaledown_start_step = attn_prior_scaledown_start_step
         self.frozen_model.enc_dec_model.attn_prior_end_step = attn_prior_end_step
+
+        alignment_decoder_layerids = cfg.get('alignment_decoder_layerids', list(range(0, 12)))
         self.frozen_model.enc_dec_model.return_all_crossattention_probs = return_all_crossattention_probs
+        if return_all_crossattention_probs:
+            # if true, requires to declare alignment_decoder_layerids for
+            # nemo.collections.nlp.modules.common.megatron.token_level_encoder_decoder.MegatronTokenLevelEncoderDecoderSpeechLLMModule.
+            self.frozen_model.enc_dec_model.alignment_decoder_layerids = alignment_decoder_layerids
```

Comment:
This is required if we are only doing cross-attention on certain layers. This doesn't have anything to do with return_all_crossattention_probs

Reply:
agreed. In our implementation,  `nemo.collections.nlp.modules.common.megatron.token_level_encoder_decoder.MegatronTokenLevelEncoderDecoderSpeechLLMModule` has to define this member variable, otherwise, it would fail in [this line]( https://github.com/blisc/NeMo/blob/speechllm_tts_2402/nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py#L975) because `self.alignment_decoder_layerids` is not defined. I can change to ensure this variable is defined.